azkaban-common/src/main/java/azkaban/storage/StorageManager.java

```diff
@@ -114,6 +119,8 @@ public void uploadProject(
       log.info(String.format("Added project metadata to DB. Meta:%s File: %s[%d bytes] URI: %s",
           metadata, localFile.getName(), localFile.length(), resourceId));
     }
+
```

Comment:
I think if upload fails, then it probably will raise an exception and the delete code will not execute.